hadoop-tools/hadoop-aws/src/main/java/org/apache/hadoop/fs/s3a/impl/RequestFactoryImpl.java

```diff
@@ -356,7 +362,7 @@ public PutObjectRequest.Builder newPutObjectRequestBuilder(String key,
     }
 
     // Set the timeout for object uploads but not directory markers.
-    if (!isDirectoryMarker) {
+    if (!isDirectoryMarker && !isCRTEnabled) {
```

Comment:
currently cannot set request level timeouts with CRT

Reply:
needs to go into the docs then